### ISSUE TITLE
Story 1 : the simplest use case

### ISSUE THREAD

**Issue by aowss** [aowss/find-a-tutor#1]
Simple matching
No registration
Single rate, single time slot, location name

```
POST /tutor
{
    "subject" : "maths",
    "rate" : 40,
    "location" : "Mississauga"
    "timeSlot" : {
        "start" : "13:00",
        "end" : "17:00"
    }
}
```

`GET /provider?subject=maths&maxRate=45&location=Mississauga&day=THURSDAY&slotStart=12:00&slotEnd=19:00`

matching :
- subject : exact match with subject
- maxRate : match if rate <= maxRate
- location : exact match with location
- start & snd : match if start and end overlap

**Comment by electrichead** [comment]
`timeslot`: I think that the GET might need to specify the duration of the engagement (e.g. 1 hour) as well as range of times (e.g. between 3PM and 7PM). This should find a tutor who is available during that range (and at this stage ignoring schedule conflicts).

The reason is that you might have a request for 3PM to 7PM and a tutor with availability from 12PM to 4PM. This should return that tutor because the engagement (1hr) fits in the overlap. Whereas if the engagement was for 2 hours, it wouldn't return this tutor.

Also, I think that `location` should be 1-to-many even in the simple case i.e. tutor would have an array of locations.

**Comment by electrichead** [comment]
I also think that we should maybe separate tutors from their "engagement intents". As in have the POST to tutors (containing `location` and `timeSlot` } separate from the POST for each of their intents (containing `rate` and `subject` ). Thoughts?

**Comment by aowss** [comment]
Add a new story that is an increment on the simple case. if you think it is 1.1, i am fine but let’s keep the simplest stories as simple as possible. we increment and it makes it easier.
many locations i feel should not be in the simplest story.

**Comment by aowss** [comment]
Yes. the tutor creation and then the adding of their intents will ultimately be separate.

**Comment by aowss** [comment]
register tutor

**Comment by aowss** [comment]
what is the pivotal tracker stuff ? are we going to use that for the stories or the stuff that is in github.
I would like to keep things in github.
or this is something else ?
